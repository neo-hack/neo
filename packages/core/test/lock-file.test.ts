import createLockFile from '../src/utils/lock-file'

import path from 'path'
import tempy from 'tempy'

const storeDir = path.join(tempy.directory(), '.store')

const lockFile = createLockFile({ storeDir })

describe('lock file', () => {
  it('read non exit lockfile should return empty', async () => {
    expect(await lockFile.read()).toMatchObject({})
  })
  it('update preset should work', async () => {
    await lockFile.updatePreset({
      neo: {
        templates: [
          {
            name: '@aiou/ts-lib-template',
          },
        ],
      },
    })
    expect(await lockFile.read()).toMatchSnapshot()
  })
  it('update templates should work', async () => {
    await lockFile.updateTemplates({
      '/@aiou/ts-lib-template/0.5.0': {},
    })
    expect(await lockFile.read()).toMatchSnapshot()
  })
  it('read templates should work', async () => {
    expect(await lockFile.readTemplates()).toMatchSnapshot()
  })
  it.todo('update packages should work')
})
