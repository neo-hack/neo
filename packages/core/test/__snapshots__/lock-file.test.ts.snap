// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lock file read templates should work 1`] = `
Array [
  Object {
    "name": "@aiou/ts-lib-template",
  },
]
`;

exports[`lock file update preset should work 1`] = `
Object {
  "presets": Object {
    "neo": Object {
      "templates": Array [
        Object {
          "name": "@aiou/ts-lib-template",
        },
      ],
    },
  },
}
`;

exports[`lock file update templates should work 1`] = `
Object {
  "presets": Object {
    "neo": Object {
      "templates": Array [
        Object {
          "name": "@aiou/ts-lib-template",
        },
      ],
    },
  },
  "templates": Object {
    "/@aiou/ts-lib-template/0.5.0": Object {},
  },
}
`;
